src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/SharedEntrypoint.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.Extension.Common
+{
+    internal sealed class SharedEntrypoint<T> where T : class, IEgressProviderOptions, new()
+    {
+        private static Stream StdInStream;
+        private static CancellationTokenSource CancelSource = new CancellationTokenSource();
+
+        private static EgressProvider _provider;
+        private static Action<ExtensionEgressPayload, T> _configureOptions;
+
+        public static async Task<int> Entrypoint(string[] args, EgressProvider provider, Action<ExtensionEgressPayload, T> configureOptions)
+        {
+            _provider = provider;
+            _configureOptions = configureOptions;
+
+            using var loggerFactory = LoggerFactory.Create(builder =>
```

Comment:
This logger factory doesn't do anything so it should be removed.

Reply:
Accidental left-over - removed.